### DISCUSSION TITLE
GeoIP timezones for v2

### DISCUSSION THREAD

**Discussion by ghost** [NamelessMC/Nameless#2128] [Ideas]
We are now able to select a timezone but it would maybe be a better idea to pick a users timezone based on its current location. It would also be cool to give the user a option to disable this feature and let him select their own timezone.

**Comment by ghost** [comment]
And maybe reflect this to the language preference.

**Comment by samerton** [comment]
As we'd need to use a third party API for obtaining IP locations, I'm not sure this will be possible sadly.

**Comment by ghost** [comment]
http://php.net/manual/en/book.geoip.php

**Comment by samerton** [comment]
I think it's better to do this using JavaScript and send that to the site instead of basing this off IP address.

**Comment by A248** [comment]
If you use a GeoIP, users with a VPN may be placed in an incorrect timezone if VPN server timezone ≠ user timezone. If this is added, it should use js as @samerton said.
